Assignment 2.Rmd

```diff
@@ -79,16 +93,109 @@ plot(g,layout=layout.fruchterman.reingold, vertex.color=VERTEX$gender)
 
 #We can change the thickness of the edge according to the number of times a particular teacher has sent a particular student to the principal.
 
-plot(g,layout=layout.fruchterman.reingold, vertex.color=VERTEX$gender, edge.width=EDGE$count)
-````
+plot(g,layout=layout.fruchterman.reingold, 
+     vertex.color=VERTEX$gender, 
+     edge.width=EDGE$count)
+```
 
 ## Part II
 
 In Part II your task is to [look up](http://igraph.org/r/) in the igraph documentation and create a graph that sizes the student vertices in terms of the number of disciplinary actions they have recieved, and the teachers in terms of the number of disciplinary actions they have given out.
 
+```{r}
+# number of disciplinary actions given by teacher
+tcount<-group_by(EDGE, from) %>%
+  summarise(sum(count))
+names(tcount)<-c("id","COUNT")
+
+#number of disciplinary actions received by student
+scount<-group_by(EDGE, to) %>%
+  summarise(sum(count))
+names(scount)<-c("id","COUNT")
+
+#combine tcount with scount for disc actions per id
+discact<-bind_rows(tcount, scount)
+VERTEX <- left_join(VERTEX, discact, by = "id")
+VERTEX <- transform(VERTEX, group = as.factor(group))
+```
+```{r}
+#Graph
+g <- graph.data.frame(EDGE, directed=TRUE, vertices=VERTEX)
+plot.igraph(g, vertex.size=VERTEX$COUNT*3,
+            edge.arrow.width=0,
+            vertex.color=VERTEX$group,
+            vertex.frame.color = NA,
+            edge.width=EDGE$count, 
+            layout=layout_nicely)
+```
+
 ## Part III
 
 Now practice with data from our class. Please create a **person-network** with the data set hudk4050-classes.csv. To create this network you will need to create a person-class matrix using the tidyr functions and then create a person-person matrix using `t()`. You will then need to plot a matrix rather than a data frame using igraph.
 
+Creating a person-class matrix:
+```{r}
+#Data wrangling
+library(tidyr)
+D3<-read.csv("HUDK4050-2018-classes.csv")
+D3<-select(D3, num_range("Q", 1:9))
+D3<-D3[,c("Q8","Q9","Q1","Q3","Q4","Q5","Q6")]
```

Comment:
I'm not sure why you have used both `select()` and matrix queries? They do the same thing.